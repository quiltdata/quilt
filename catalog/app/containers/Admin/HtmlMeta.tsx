import * as React from 'react'
import { Helmet } from 'react-helmet'

interface MetaTitleProps {
  subtitle?: string
}

function MetaTitle({ subtitle }: MetaTitleProps) {
  return (
    <Helmet>
      <title>
        {subtitle ? `${subtitle} • ` : ''}
        Admin • Quilt is a versioned data portal for AWS
      </title>
    </Helmet>
  )
}

export function UsersAndRoles() {
  return <MetaTitle subtitle="Users and Roles" />
}

export function Buckets() {
  return <MetaTitle subtitle="Buckets" />
}
