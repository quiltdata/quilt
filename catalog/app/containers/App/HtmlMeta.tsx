import * as React from 'react'
import { Helmet } from 'react-helmet'

interface MetaTitleProps {
  subtitle?: string
}

function MetaTitle({ subtitle }: MetaTitleProps) {
  return (
    <Helmet>
      <title>
        {subtitle ? `${subtitle} â€¢ ` : ''}
        Quilt is a versioned data portal for AWS
      </title>
    </Helmet>
  )
}

interface SearchProps {
  query?: string
}

export function Search({ query }: SearchProps) {
  const subtitle = query || 'Search'
  return <MetaTitle subtitle={subtitle} />
}

export function UriResolver() {
  return <MetaTitle subtitle="Resolve a Quilt package URI" />
}
