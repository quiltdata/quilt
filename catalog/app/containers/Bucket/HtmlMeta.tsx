import * as R from 'ramda'
import * as React from 'react'
import { Helmet } from 'react-helmet'

interface MetaTitleProps {
  bucket: string
  subtitle?: string
}

function MetaTitle({ bucket, subtitle }: MetaTitleProps) {
  return (
    <Helmet>
      {subtitle ? (
        <title>
          {subtitle} • {bucket} • Quilt
        </title>
      ) : (
        <title>{bucket} • Quilt</title>
      )}
    </Helmet>
  )
}

interface BaseProps {
  bucket: string
}

interface RootProps extends BaseProps {}

export function Root({ bucket }: RootProps) {
  return <MetaTitle bucket={bucket} />
}

interface FileProps extends BaseProps {
  path: string
}

export function File({ bucket, path }: FileProps) {
  const subtitle = path || 'Files'
  return <MetaTitle bucket={bucket} subtitle={subtitle} />
}

export const Dir = File

export const Overview = Root

interface PackageListProps extends BaseProps {}

export function PackageList({ bucket }: PackageListProps) {
  return <MetaTitle bucket={bucket} subtitle="Package list" />
}

interface RevisionsProps extends BaseProps {
  name: string
}

export function Revisions({ bucket, name }: RevisionsProps) {
  return <MetaTitle bucket={bucket} subtitle={name} />
}

interface PackageTreeProps extends BaseProps {
  name: string
  path: string
  revision: string
}

export function PackageTree({ bucket, name, path, revision }: PackageTreeProps) {
  const revisionShortened = R.take(10, revision)
  const subtitle = `${name}@${revisionShortened}/${path}`
  return <MetaTitle bucket={bucket} subtitle={subtitle} />
}

interface QueriesProps extends BaseProps {}

export function Queries({ bucket }: QueriesProps) {
  return <MetaTitle bucket={bucket} subtitle="Queries" />
}

interface SearchProps extends BaseProps {
  query: string
}

export function Search({ bucket, query }: SearchProps) {
  return <MetaTitle bucket={bucket} subtitle={query} />
}
