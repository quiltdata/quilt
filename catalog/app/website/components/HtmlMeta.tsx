import * as React from 'react'
import { Helmet } from 'react-helmet'

interface LandingMetaTitleProps {
  subtitle?: string
}

function LandingMetaTitle({ subtitle }: LandingMetaTitleProps) {
  return (
    <Helmet>
      <title>
        {subtitle ? `${subtitle} â€¢ ` : ''}
        Quilt is a versioned data portal for AWS
      </title>
    </Helmet>
  )
}

export function Landing() {
  return <LandingMetaTitle />
}

export const OpenLanding = Landing

export function OpenProfile() {
  return <LandingMetaTitle subtitle="Profile" />
}

export function Install() {
  return <LandingMetaTitle subtitle="Install" />
}

export function About() {
  return <LandingMetaTitle subtitle="About" />
}

export function Personas() {
  return <LandingMetaTitle subtitle="Personas" />
}

export function Product() {
  return <LandingMetaTitle subtitle="Product" />
}
