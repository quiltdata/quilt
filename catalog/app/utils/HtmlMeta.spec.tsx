import * as React from 'react'
import renderer from 'react-test-renderer'

import * as HtmlMeta from './HtmlMeta'

describe('utils/HtmlMeta', () => {
  describe('HtmlMeta', () => {
    it('should render base title', () => {
      const tree = renderer.create(<HtmlMeta.Meta />).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('should render base title plus prefix title when prefix is string', () => {
      const tree = renderer.create(<HtmlMeta.Meta>Specific title</HtmlMeta.Meta>).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('should render plus divided prefix titles when prefix is array of strings', () => {
      const tree = renderer
        .create(
          <HtmlMeta.Meta>{['Specific title #1', 'Specific title #2']}</HtmlMeta.Meta>,
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  describe('getTitle', () => {
    it('returns base when no arguments', () => {
      expect(HtmlMeta.getTitle('Base title')).toBe('Base title')
    })

    it('returns base plus prefix when prefix is string', () => {
      expect(HtmlMeta.getTitle('Base title', 'Specific title')).toBe(
        'Specific title • Base title',
      )
    })

    it('returns base plus divided prefix when prefix is array of strings', () => {
      expect(
        HtmlMeta.getTitle('Base title', ['Specific title #1', 'Specific title #2']),
      ).toBe('Specific title #1 • Specific title #2 • Base title')
    })
  })
})
