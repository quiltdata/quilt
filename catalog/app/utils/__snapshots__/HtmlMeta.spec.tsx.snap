// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/HtmlMeta HtmlMeta should render base title 1`] = `
<title>
  Quilt is a versioned data portal for AWS
</title>
`;

exports[`utils/HtmlMeta HtmlMeta should render base title plus prefix title when prefix is string 1`] = `
<title>
  Specific title • Quilt is a versioned data portal for AWS
</title>
`;

exports[`utils/HtmlMeta HtmlMeta should render plus divided prefix titles when prefix is array of strings 1`] = `
<title>
  Specific title #1 • Specific title #2 • Quilt is a versioned data portal for AWS
</title>
`;
